core/src/test/java/org/apache/iceberg/TestBatchScans.java

```diff
@@ -51,27 +48,27 @@ public void testDataTableScan() {
     BatchScan scan = table.newBatchScan();
 
     List<ScanTask> tasks = planTasks(scan);
-    Assert.assertEquals("Expected 2 tasks", 2, tasks.size());
+    assertThat(tasks).hasSize(2);
 
     FileScanTask t1 = tasks.get(0).asFileScanTask();
-    Assert.assertEquals("Task file must match", t1.file().path(), FILE_A.path());
-    V1Assert.assertEquals("Task deletes size must match", 0, t1.deletes().size());
-    V2Assert.assertEquals("Task deletes size must match", 1, t1.deletes().size());
+    assertThat(FILE_A.path()).as("Task file must match").isEqualTo(t1.file().path());
+    V1Assert.assertHasSize("Task deletes size must match", 0, t1.deletes());
+    V2Assert.assertHasSize("Task deletes size must match", 1, t1.deletes());
```

Comment:
I think changing this should be done separately from migrating away from JUnit4. That means that we can consider making `TableAssertions` more fluent and more like AssertJ assertions, but that shouldn't be part of this PR

Reply:
Got it, thank you. I will create another PR for that migration. For now, keep the original one.